async_tcp_server/src/main.rs

```diff
@@ -0,0 +1,37 @@
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tokio::net::TcpListener;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let listener = TcpListener::bind("127.0.0.1:8080").await?;
+
+    loop {
+        let (mut socket, addr) = listener.accept().await?;
+        println!("got connection from {:?}", addr);
+
+        tokio::spawn(async move {
+            let mut buf = [0; 10];
+
+            loop {
+                let n = match socket.read(&mut buf).await {
+                    // socket closed
+                    Ok(n) if n == 0 => break,
```

Comment:
Noting down here that apparently tokio has no support to just monitor the connection without reading or writing, i.e. checking POLLRDHUP on Linux.